### PR TITLE
Bagu changes

### PR DESCRIPTION
**Pull request by ghost**
Fixed mainly on entity and chunk generator

### CODE REVIEW

**Comment by williewillus** [comment]
you have a ton of needless whitespace changes. please keep the whitespace consistent with the existing whitespace of the file
(if there was a mix of spacing, make it consistent, picking the one that appeared more often in that file)

File: src/main/java/twilightforest/TFEventListener.java [hunk 1]
```diff
@@ -630,7 +633,7 @@ public static void playerLogsIn(PlayerEvent.PlayerLoggedInEvent event) {
 	@SubscribeEvent
 	public static void playerPortals(PlayerEvent.PlayerChangedDimensionEvent event) {
 		if (!event.getPlayer().world.isRemote && event.getPlayer() instanceof ServerPlayerEntity) {
-			if (event.toDim == TFConfig.dimension.dimensionID) {
```
Comment (Andromander) [inline src/main/java/twilightforest/TFEventListener.java:633]: Incorrect spacing.

File: src/main/java/twilightforest/client/model/entity/ModelTFGoblinKnightLower.java [hunk 2]
```diff
@@ -69,17 +69,8 @@ public ModelTFGoblinKnightLower() {
...
-			this.renderTunic(stack, builder, light, overlay, red, green, blue, scale);
-		}
-	}
-
-	/**
-	 * Renders the tunic, if we're wearing armor
-	 */
-	public void renderTunic(MatrixStack stack, IVertexBuilder builder, int light, int overlay, float red, float green, float blue, float scale) {
 		this.tunic.render(stack, builder, light, overlay, red, green, blue, scale);
+
```
Comment (Andromander) [inline src/main/java/twilightforest/client/model/entity/ModelTFGoblinKnightLower.java]: Unnecessary whitespace

File: src/main/java/twilightforest/TwilightForestMod.java [hunk 3]
```diff
@@ -58,7 +59,7 @@
 	// odd one out, as armor textures are a stringy mess at present
 	public static final String ARMOR_DIR  = ID + ":textures/armor/";
 
-	public static final String ENFORCED_PROGRESSION_RULE = "tfEnforcedProgression";
+    public static final GameRules.RuleKey<GameRules.BooleanValue> ENFORCED_PROGRESSION_RULE = GameRules.register("tfEnforcedProgression", GameRules.BooleanValue.create(true));
```
Comment (Andromander) [inline src/main/java/twilightforest/TwilightForestMod.java]: Incorrect spacing. Must use Tabs.

File: src/main/java/twilightforest/client/model/entity/ModelTFGoblinKnightLower.java [hunk 4]
```diff
@@ -125,5 +116,9 @@ public void setAngles(T entity, float limbSwing, float limbSwingAmount, float ag
 		this.bipedLeftArm.rotateAngleZ -= MathHelper.cos(ageInTicks * 0.09F) * 0.05F + 0.05F;
 		this.bipedRightArm.rotateAngleX += MathHelper.sin(ageInTicks * 0.067F) * 0.05F;
 		this.bipedLeftArm.rotateAngleX -= MathHelper.sin(ageInTicks * 0.067F) * 0.05F;
+
+
+        this.tunic.showModel = entity.hasArmor();
+
```
Comment (Andromander) [inline src/main/java/twilightforest/client/model/entity/ModelTFGoblinKnightLower.java]: Spacing here appears to use Space not Tab.

File: src/main/java/twilightforest/client/model/entity/ModelTFGoblinKnightUpper.java [hunk 5]
```diff
@@ -128,19 +128,11 @@ public ModelTFGoblinKnightUpper() {
...
 	public void render(MatrixStack stack, IVertexBuilder builder, int light, int overlay, float red, float green, float blue, float scale) {
-		this.shield.showModel = entity.hasShield();
 		super.render(stack, builder, light, overlay, red, green, blue, scale);
 
-		if (entity.hasArmor()) {
-			this.renderBreastplate(stack, builder, light, overlay, red, green, blue, scale);
-		}
-	}
+        this.breastplate.render(stack, builder, light, overlay, red, green, blue, scale);
+
```
Comment (Andromander) [inline src/main/java/twilightforest/client/model/entity/ModelTFGoblinKnightUpper.java]: Incorrect spacing. Also, unnecessary whitespace.

File: src/main/java/twilightforest/client/model/entity/ModelTFYetiAlpha.java [hunk 6]
```diff
@@ -108,16 +109,10 @@ private void addPairHorns(float height, float zangle) {
...
-	@Override
-	public void render(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale) {
-		this.setRotationAngles(entity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale);
 
-		this.bipedBody.render(scale);
-		this.bipedRightLeg.render(scale);
-		this.bipedLeftLeg.render(scale);
+	@Override
+    protected Iterable<ModelRenderer> getBodyParts() {
+        return ImmutableList.of(this.bipedBody, this.bipedRightLeg, this.bipedLeftLeg);
```
Comment (Andromander) [inline src/main/java/twilightforest/client/model/entity/ModelTFYetiAlpha.java]: Incorrect spacing.

File: src/main/java/twilightforest/client/renderer/entity/RenderTFSnowGuardian.java [hunk 7]
```diff
@@ -11,7 +12,7 @@
 
 	public RenderTFSnowGuardian(EntityRendererManager manager, M model) {
 		super(manager, model, 0.25F, "textures/entity/zombie/zombie.png");
-		this.addLayer(new BipedArmorLayer<>(this));
+        this.addLayer(new BipedArmorLayer<>(this, new ModelTFSnowGuardian<>(), new ModelTFSnowGuardian<>()));
```
Comment (Andromander) [inline src/main/java/twilightforest/client/renderer/entity/RenderTFSnowGuardian.java]: Incorrect spacing.

File: src/main/java/twilightforest/entity/EntityTFPinchBeetle.java [hunk 8]
```diff
@@ -137,4 +129,14 @@ public boolean canRiderInteract() {
...
+
+    @Override
+    public EntitySize getSize(Pose p_213305_1_) {
+
+        if (!this.getPassengers().isEmpty()) {
+            return EntitySize.flexible(1.9F, 2.0F);
+        } else {
+            return super.getSize(p_213305_1_);
+        }
+    }
```
Comment (Andromander) [inline src/main/java/twilightforest/entity/EntityTFPinchBeetle.java]: Spacing here appears to use Space not Tab.

File: src/main/java/twilightforest/entity/EntityTFTowerGhast.java [hunk 9]
```diff
@@ -309,15 +312,21 @@ protected void spitFireball() {
...
+		int i = world.getWorld().isThundering() ? world.getNeighborAwareLightSubtracted(blockPos, 10) : world.getLight(blockPos);
+		return i <= p_223323_2_.nextInt(8);
 	}
 
+	public boolean isNotColliding(IWorldReader p_205019_1_) {
+		return p_205019_1_.intersectsEntities(this) && !p_205019_1_.containsAnyLiquid(this.getBoundingBox());
+	}
+
+
+
```
Comment (Andromander) [inline src/main/java/twilightforest/entity/EntityTFTowerGhast.java]: Lots of unneccesary whitespace here.

File: src/main/java/twilightforest/entity/EntityTFYeti.java [hunk 10]
```diff
@@ -17,202 +18,213 @@
...
+
+    @Override
+    protected SoundEvent getDeathSound() {
+        return TFSounds.ALPHAYETI_DIE;
+    }
+
+    @Override
+    public ResourceLocation getLootTable() {
+        return LOOT_TABLE;
+    }
```
Comment (Andromander) [inline src/main/java/twilightforest/entity/EntityTFYeti.java]: This diff indicates almost all the class is different due to incorrect spacing and is marked as a "large diff". Use Tabs.

File: src/main/java/twilightforest/entity/boss/EntityTFMinoshroom.java [hunk 11]
```diff
@@ -132,7 +132,7 @@ public boolean canDespawn(double p_213397_1_) {
 	@Override
 	public void checkDespawn() {
 		if (world.getDifficulty() == Difficulty.PEACEFUL) {
-			if (hasHome()) {
+            if (detachHome()) {
```
Comment (Andromander) [inline src/main/java/twilightforest/entity/boss/EntityTFMinoshroom.java]: Incorrect spacing.

File: src/main/java/twilightforest/entity/boss/EntityTFUrGhastFireball.java [hunk 12]
```diff
@@ -28,12 +31,23 @@ protected void onImpact(RayTraceResult result) {
...
+    public void shoot(double p_70186_1_, double p_70186_3_, double p_70186_5_, float p_70186_7_, float p_70186_8_) {
+        Vec3d vec3d = (new Vec3d(p_70186_1_, p_70186_3_, p_70186_5_)).normalize().add(this.rand.nextGaussian() * (double) 0.0075F * (double) p_70186_8_, this.rand.nextGaussian() * (double) 0.0075F * (double) p_70186_8_, this.rand.nextGaussian() * (double) 0.0075F * (double) p_70186_8_).scale((double) p_70186_7_);
+        this.setMotion(vec3d);
+        float f = MathHelper.sqrt(horizontalMag(vec3d));
+        this.rotationYaw = (float) (MathHelper.atan2(vec3d.x, p_70186_5_) * (double) (180F / (float) Math.PI));
+        this.rotationPitch = (float) (MathHelper.atan2(vec3d.y, (double) f) * (double) (180F / (float) Math.PI));
+        this.prevRotationYaw = this.rotationYaw;
+        this.prevRotationPitch = this.rotationPitch;
+    }
+
```
Comment (Andromander) [inline src/main/java/twilightforest/entity/boss/EntityTFUrGhastFireball.java:50]: Incorrect spacing. `shoot` method shows to be using Space instead of Tab.

File: src/main/java/twilightforest/entity/boss/EntityTFYetiAlpha.java [hunk 13]
```diff
@@ -331,7 +331,7 @@ public boolean canDespawn(double p_213397_1_) {
 	@Override
 	public void checkDespawn() {
 		if (world.getDifficulty() == Difficulty.PEACEFUL) {
-			if (hasHome()) {
+            if (detachHome()) {
```
Comment (Andromander) [inline src/main/java/twilightforest/entity/boss/EntityTFYetiAlpha.java]: Incorrect spacing.

File: src/main/java/twilightforest/entity/passive/EntityTFTinyBird.java [hunk 14]
```diff
@@ -146,7 +146,7 @@ public void tick() {
 		super.tick();
 		// while we are flying, try to level out somewhat
 		if (!this.isBirdLanded()) {
-			this.motionY *= 0.6000000238418579D;
+            this.setMotion(this.getMotion().mul(1.0F, 0.6000000238418579D, 1.0F));
```
Comment (Andromander) [inline src/main/java/twilightforest/entity/passive/EntityTFTinyBird.java]: Incorrect spacing.

File: src/main/java/twilightforest/util/ParticleHelper.java [hunk 15]
```diff
@@ -1,18 +1,19 @@
...
 
 //TODO: Fix methods
 public class ParticleHelper {
 
-	public static void spawnParticles(World world, BlockPos pos, ParticleTypes type, int count, double speed, int... particleArgs) {
-		world.addParticle(type, pos.getX() + 0.5, pos.getY() + 0.5, pos.getZ() + 0.5, count, 0.5, 0.5, 0.5, speed, particleArgs);
-	}
+    public static void spawnParticles(World world, BlockPos pos, IParticleData type, int count, double speed, int... particleArgs) {
+        world.addParticle(type, pos.getX() + 0.5, pos.getY() + 0.5, pos.getZ() + 0.5, 0.5, 0.5, 0.5);
+    }
```
Comment (Andromander) [inline src/main/java/twilightforest/util/ParticleHelper.java]: Incorrect spacing.

File: src/main/java/twilightforest/world/ChunkGeneratorTwilightForest.java [hunk 16]
```diff
@@ -4,59 +4,56 @@
...
-		//this.scaleNoise = new NoiseGeneratorOctaves(rand, 10);
-		//this.forestNoise = new NoiseGeneratorOctaves(rand, 8);
+        this.noiseGen4 = new OctavesNoiseGenerator(this.randomSeed, 4, 0);
+        //this.scaleNoise = new OctavesNoiseGenerator(rand, 10);
+        //this.forestNoise = new OctavesNoiseGenerator(rand, 8);
 	}
 
 	@Override
 	public void decorate(WorldGenRegion region) {
+        super.decorate(region);
```
Comment (Andromander) [inline src/main/java/twilightforest/world/ChunkGeneratorTwilightForest.java]: Incorrect spacing.

File: src/main/java/twilightforest/world/ChunkGeneratorTwilightVoid.java [hunk 17]
```diff
@@ -28,17 +22,17 @@ public void decorate(WorldGenRegion region) {
 //		rand.setSeed(getSeed(x, z));
 
 		ChunkBitArray data = new ChunkBitArray();
-		func_222529_a(x, z, data);
+        //func_222529_a(x, z, data);
```
Comment (Andromander) [inline src/main/java/twilightforest/world/ChunkGeneratorTwilightVoid.java]: Spacing here appears to be altered to Space.

**Review by Andromander — CHANGES_REQUESTED** [review]
A lot of classes in this Pull Request use incorrect spacing, using Space instead of Tab. For the sake of brevity, any classes I've commented on noting spacing has problematic indenting throughout, not just the one line. Any lines that appear to highlight nothing as a difference, deletion and addition despite being the same, and dragging a cursor over empty space revealing Space indicates that it is not using Tab.
Make sure project settings are set to use Tab instead of Space (usually under Code Style in Settings), and check over formatting to make sure the classes are updated as such.

As for the actual changes themselves, they're fine as they are. There are some extra whitepace that could be removed.

File: src/main/java/twilightforest/entity/EntityTFYeti.java [hunk 18]
```diff
@@ -17,202 +18,213 @@
...
-	@Override
-	protected SoundEvent getDeathSound() {
-		return TFSounds.ALPHAYETI_DIE;
-	}
-
-	@Override
-	public ResourceLocation getLootTable() {
-		return LOOT_TABLE;
-	}
+    public static final ResourceLocation LOOT_TABLE = TwilightForestMod.prefix("entities/yeti");
```
Comment (Andromander) [inline src/main/java/twilightforest/entity/EntityTFYeti.java]: Class is still entirely changed due to incorrect spacing. Note that the diff says the content was deleted and then added back in.

**Review by Andromander — COMMENTED** [review]
Almost there, just one class is still using improper spacing. Should be good to go afterwards.

File: src/main/java/twilightforest/TFEventListener.java [hunk 19]
```diff
@@ -680,22 +686,23 @@ public static void onServerDisconnect(PlayerEvent.PlayerLoggedOutEvent event) {
 	public static void worldLoaded(WorldEvent.Load event) {
 		World world = event.getWorld().getWorld();
 
-		if (!world.isRemote() && !world.getGameRules().get(TwilightForestMod.ENFORCED_PROGRESSION_RULE)) {
+		if (!world.isRemote() && world.getGameRules().get(TwilightForestMod.ENFORCED_PROGRESSION_RULE) != null) {
```
Comment (Tamaized) [inline src/main/java/twilightforest/TFEventListener.java]: Just a general question.. is this actually proper..? Is the game rule always null when false? Just seems weird.

**Review by Andromander — APPROVED** [review]
